### PR TITLE
Refactor/heartbeat config

### PR DESCRIPTION
**Pull request by brennan**
- Refactors Heartbeat initialization - new Heartbeat instances are automatically configured when a customer sends a "Video Playback Started" event
- Updates appropriate tests
- This PR does not update event and property mappings, which Ladan and I agree on earlier today - these will be updated in a separate PR

### CODE REVIEW

File: src/main/java/com/segment/analytics/android/integrations/adobeanalytics/AdobeIntegration.java [hunk 1]
```diff
@@ -399,9 +386,49 @@ public void reset() {
...
+        config = new MediaHeartbeatConfig();
+
+        config.trackingServer = heartbeatTrackingServer;
+        if (properties.get("channel") != null) {
+          config.channel = properties.getString("channel");
+        } else {
+          config.channel = "";
+        }
+        // default app version to 0.0 if not otherwise present b/c Adobe requires this value
+        if (!isNullOrEmpty(context.getPackageName())) {
```
Comment (f2prateek) [inline src/main/java/com/segment/analytics/android/integrations/adobeanalytics/AdobeIntegration.java]: We could just do save this once during initialization

File: src/main/java/com/segment/analytics/android/integrations/adobeanalytics/AdobeIntegration.java [hunk 2]
```diff
@@ -76,10 +78,14 @@ public String key() {
 
   MediaHeartbeatConfig config;
   private MediaHeartbeat heartbeat;
+  HeartbeatFactory heartbeatFactory;
```
Comment (f2prateek) [inline src/main/java/com/segment/analytics/android/integrations/adobeanalytics/AdobeIntegration.java]: These fields should be `final` unless it's required for testing.

File: src/main/java/com/segment/analytics/android/integrations/adobeanalytics/AdobeIntegration.java [hunk 3]
```diff
@@ -76,10 +78,14 @@ public String key() {
...
+  String heartbeatTrackingServer;
```
Comment (f2prateek) [inline src/main/java/com/segment/analytics/android/integrations/adobeanalytics/AdobeIntegration.java]: Also for a style thing, these should also should be organized by their qualifier. Right now constants are being mixed in order with member variables.

File: src/main/java/com/segment/analytics/android/integrations/adobeanalytics/AdobeIntegration.java (hunk 1 above)
Comment (brennan) [reply src/main/java/com/segment/analytics/android/integrations/adobeanalytics/AdobeIntegration.java]: You are totally right. Just fixed it.

File: src/main/java/com/segment/analytics/android/integrations/adobeanalytics/AdobeIntegration.java (hunk 3 above)
Comment (brennan) [reply src/main/java/com/segment/analytics/android/integrations/adobeanalytics/AdobeIntegration.java]: Just did this. Thanks!

File: src/main/java/com/segment/analytics/android/integrations/adobeanalytics/AdobeIntegration.java [hunk 4]
```diff
@@ -80,6 +79,7 @@ public String key() {
   private MediaHeartbeat heartbeat;
   HeartbeatFactory heartbeatFactory;
   String heartbeatTrackingServer;
+  String packageName;
```
Comment (f2prateek) [inline src/main/java/com/segment/analytics/android/integrations/adobeanalytics/AdobeIntegration.java]: should be final

**Pull request merged**